lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -621,14 +555,21 @@ bool TypeLoweringVisitor::lowerProducer(
     ArrayAttr loweredAttrs = filterAnnotations(context, oldAnno, srcType, field,
                                                needsSym, loweredSymName);
     auto *newOp = clone(field, loweredAttrs);
+
+    // If this is a ground type, record a rename to it.
+    if (field.type.isGround() && innerSymAttr) {
+      auto module = op->getParentOfType<FModuleOp>();
+      innerRefRenames[hw::InnerRefAttr::get(module.getNameAttr(), innerSymAttr)]
+          .push_back(OpAnnoTarget(newOp));
```

Comment:
Since the rename map is used only for `HierPathOp`, what if we check for `circt.nonlocal` annotation here and insert the lowered field into the rename map only if the ground type has a nonlocal annotation.